pkg/lib/cockpit-upload-helper.ts

```diff
@@ -93,7 +139,13 @@ export async function upload(
 
     try {
         debug('starting file send', contents);
-        const reader = contents.stream().getReader({ mode: 'byob' }); // byob to choose the block size
+        let reader;
+        try {
+            reader = new ByobReader(contents.stream().getReader({ mode: 'byob' }));
+        } catch {
```

Comment:
This is a very wide catch -- would be good to check the error, and also add a comment pointing to caniuse -- we want to remove this in a year or two?

Reply:
I tried looking for an issue tracking the implementation of what we need in webkit but that seems hopeless. 

The codebase does seem to implement [ReadableByteStreamController and ReadableStreamBYOBReader](https://github.com/WebKit/WebKit/blob/5a0fff2319c5d09325cdb3a17efdf2bff6d6662d/Source/WebCore/Modules/streams/ReadableStreamBYOBReader.js). So maybe what we need is implemented soon?

Reply:
Can't you check for the existence of the BYOB function, instead of using a blanket try/catch, like this instead?

```js
if (typeof ReadableStreamBYOBReader !== 'undefined') {
  /* BOYB path */
} else {
  /* fallback for WebKit */
}
```

or

```js
if (typeof ReadableStreamBYOBReader === 'function') {
  /* BOYB path */
} else {
  /* fallback for WebKit */
}

```
`typeof` for functions would return either `"function"` or `"undefined"` _as strings_, hence the string check.

I checked in Firefox, Chromium, and GNOME Web consoles. `(typeof ReadableStreamBYOBReader !== 'undefined')` and `(typeof ReadableStreamBYOBReader === 'function')` both return `true` in Firefox and Chromium; it's `false` in WebKit.

Reply:
I like @garrett's suggestion.